Not specifying the quantum value for assets in the genesis file causes crash

If an asset is defined in the genesis file without a quantum value, it is defaulted to a value of 0 which will cause a crash when that asset is first snapshotted.

```
pete@VegaNode:~/.vegacapsule/testnet/logs/testnet-nodeset-validators-1-validator$ more vega-validator-1.stderr-2024-03-28T11\:40\:04Z.log
panic: decimal division by 0

goroutine 59765 [running]:
github.com/shopspring/decimal.Decimal.QuoRem({0xc005375620, 0x0}, {0xc00042f600, 0x1}, 0x10)
        /home/pete/go/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:590 +0x2bc
github.com/shopspring/decimal.Decimal.DivRound({0xc005375620?, 0x0?}, {0xc00042f600?, 0x0?}, 0x10)
        /home/pete/go/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:632 +0x4d
github.com/shopspring/decimal.Decimal.Div(...)
        /home/pete/go/pkg/mod/github.com/vegaprotocol/decimal@v1.3.1-uint256/decimal.go:577
code.vegaprotocol.io/vega/core/fee.(*FeesStats).ToProto(0xc0033d80c0, {0xc0039e8a68, 0x5}, {0xc00042f600?, 0x2?})
        /home/pete/vega/performance/vega/core/fee/rebate_stats.go:215 +0xa7e
code.vegaprotocol.io/vega/core/fee.(*Engine).GetState(...)
        /home/pete/vega/performance/vega/core/fee/engine.go:103
code.vegaprotocol.io/vega/core/execution/future.(*Market).GetState(0xc0110c2000)
        /home/pete/vega/performance/vega/core/execution/future/market_snapshot.go:339 +0x97f
code.vegaprotocol.io/vega/core/execution.(*Engine).marketsStates(0xc0009a0dc0)
        /home/pete/vega/performance/vega/core/execution/engine_snapshot.go:50 +0x31f
code.vegaprotocol.io/vega/core/execution.(*Engine).serialise(0xc0009a0dc0)
        /home/pete/vega/performance/vega/core/execution/engine_snapshot.go:286 +0x2a
code.vegaprotocol.io/vega/core/execution.(*Engine).GetState(0x515cec0?, {0xc0009995c0?, 0xc00445e900?})
        /home/pete/vega/performance/vega/core/execution/engine_snapshot.go:353 +0x13
code.vegaprotocol.io/vega/core/snapshot.gatherState(0xc0002c4c80, 0xc0030eaf98?, 0xc0030eafd0?, 0xc00455a8f8, 0x6294568?)
        /home/pete/vega/performance/vega/core/snapshot/worker.go:65 +0x335
code.vegaprotocol.io/vega/core/snapshot.(*Engine).snapshotNow.func2()
        /home/pete/vega/performance/vega/core/snapshot/engine.go:561 +0x30
created by code.vegaprotocol.io/vega/core/snapshot.(*Engine).snapshotNow in goroutine 290
        /home/pete/vega/performance/vega/core/snapshot/engine.go:560 +0x372
```

This only impacts people who are creating their own networks (such as via vegacapsule for testing purposes). There is no impact on mainnet as the assets are proposed at runtime and not defined at startup.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -44,6 +44,7 @@
 - [11010](https://github.com/vegaprotocol/vega/issues/11010) - Release bond in spot market when market is terminated. 
 - [11038](https://github.com/vegaprotocol/vega/issues/11038) - Simplify price ranges state. 
 - [11042](https://github.com/vegaprotocol/vega/issues/11042) - Add missing value to enum.
+- []
 
 ## 0.75.0
 
@@ -115,6 +116,7 @@
 - [10904](https://github.com/vegaprotocol/vega/issues/10904) - Balance changes API should validate date range properly.
 - [10911](https://github.com/vegaprotocol/vega/issues/10911) - Fix `unregistration` of spot market state var.
 - [10915](https://github.com/vegaprotocol/vega/issues/10915) - Fix duplicate deposit entries, add new `STATUS_DUPLICATE_REJECTED` status.
+- [11041](https://github.com/vegaprotocol/vega/issues/11041) - Add validation for genesis asset quantum. 
 
 ## 0.74.3
 

--- a/cmd/vega/commands/verify/genesis.go
+++ b/cmd/vega/commands/verify/genesis.go
@@ -90,8 +90,8 @@ func verifyAssets(r *reporter, assets map[string]asset) {
 	}
 
 	for k, v := range assets {
-		if _, failed := num.UintFromString(v.Quantum, 10); failed {
-			r.Err("app_state.assets[%s].quantum not a valid number: %s", k, v.Quantum)
+		if n, failed := num.UintFromString(v.Quantum, 10); failed || n.IsNegative() || n.IsZero() {
+			r.Err("app_state.assets[%s].quantum not a valid positive number: %s", k, v.Quantum)
 		}
 
 		switch {

--- a/core/protocol/upon_genesis.go
+++ b/core/protocol/upon_genesis.go
@@ -70,6 +70,10 @@ func (svcs *allServices) loadAsset(ctx context.Context, id string, v *proto.Asse
 		return fmt.Errorf("error instanciating asset %v", err)
 	}
 
+	if !rawAsset.Quantum.IsPositive() {
+		svcs.log.Panic("invalid asset found - asset quantum must be positive", logging.String("quantum", rawAsset.Quantum.String()))
+	}
+
 	asset, err := svcs.assets.Get(aid)
 	if err != nil {
 		return fmt.Errorf("unable to get asset %v", err)

